tools/test.py

```diff
@@ -1713,25 +1655,32 @@ def Main():
         print "Could not create the temporary directory", options.temp_dir
         sys.exit(1)
 
-  if options.report:
-    PrintReport(all_cases)
-
-  result = None
-  def DoSkip(case):
-    # A list of tests that should be skipped can be provided. This is
-    # useful for tests that fail in some environments, e.g., under coverage.
-    if options.skip_tests != [""]:
-        if [ st for st in options.skip_tests if st in case.case.file ]:
-            return True
-    if SKIP in case.outcomes or SLOW in case.outcomes:
+  def should_keep(case):
+    if any([ (s in case.file) for s in options.skip_tests ]):
+      return False
+    elif SKIP in case.outcomes:
+      return False
+    elif options.flaky_tests == SKIP and {SLOW, FLAKY} & case.outcomes:
+      return False
+    else:
       return True
-    return FLAKY in case.outcomes and options.flaky_tests == SKIP
-  cases_to_run = [ c for c in all_cases if not DoSkip(c) ]
+
+  cases_to_run = list(filter(should_keep, all_cases))
```

Comment:
Is the explicit `list` construction is to support Python 3.x?

Reply:
ask.
It can wait for a 2x3 PR.